linked_list/LinkedList.java

```diff
@@ -0,0 +1,217 @@
+/*
+* Following is the code for Linked List
+* T defines the type of Linked List to be created
+*/
+
+import java.util.NoSuchElementException;
+
+// Class to create a Node that contains data and pointer to next Node 
+class Node<T> {
+    T data;     // Data of type T
+    Node<T> next;     // Pointer to next Node
+
+    // Constructor to create a Node
+    public Node(T input) {  
+        this.data = input;
+        this.next = null;
+    }
+}
+
+// Class containing methods to be performed on Linked List
+class LinkedList_<T> {
```

Comment:
> We do not need this class at all. It can be merged in LinkedList class, wherein the main method can be static and initialize an object of the same class and do the talking.

```java
public class LinkedList<T> {
    ...
    // Methods related to linked list class
    ...

    public static void main(String[] args) {
        LinkedList<Object> list = new LinkedList<Object>();
        ...  // Other operations
    }
}
```